cheroot/workers/threadpool.py

```diff
@@ -208,10 +219,10 @@ def shrink(self, amount):
         """Kill off worker threads (not below self.min)."""
         # Grow/shrink the pool if necessary.
         # Remove any dead threads from our list
-        for t in self._threads:
-            if not t.isAlive():
-                self._threads.remove(t)
-                amount -= 1
+        amount -= len(self._pending_shutdowns)
+        self._clear_dead_threads()
```

Comment:
Maybe it's easier to return the amount of cleared threads and use it in that substitution instead? It seems a more consistent way to go...

Reply:
But we're not specifically interested in how many threads are dead at this point, we're only interested how many threads are scheduled to be shutdown. And that figure can also include threads which have already terminated.